### PR TITLE
Add unittest file for basic:label

### PR DESCRIPTION
**Pull request by devdekunle**
Write a new unittest file for basic:label testing the three functions, label_ops_set, label_ops_post and label_ops_pre

### CODE REVIEW

**Comment by devdekunle** [comment]
@bluca please kindly review. i have fixed up the commits and implemented changes

**Review by bluca — APPROVED** [review]

File: src/basic/label.c [hunk 1]
```diff
@@ -28,3 +28,6 @@ int label_ops_post(int dir_fd, const char *path) {
 
         return label_ops->post(dir_fd, path);
 }
+void label_ops_reset(void) {
```
Comment (keszybz) [inline src/basic/label.c:32]: Newline missing here.

File: src/test/test-label.c [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        if (!path_is_safe(path))
+                return -ENOTDIR;
+
+        /* assume a case where a specific label isn't allowed */
+        if (path_equal(path, "/restricted_directory"))
+                return -EACCES;
+        return 0;
+}
+static int pre_labelling_func(int dir_fd, const char *path, mode_t mode) {
```
Comment (keszybz) [inline src/test/test-label.c:35]: Newline missing here.

File: src/test/test-label.c [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+static int check_path(int dir_fd, const char *path) {
+        assert(dir_fd >= 0);
+        assert(path);
+
+        if (!is_path(path) || !path_is_valid(path) || isempty(path))
+                return -EINVAL;
+
+        if (strlen(path) > 40)
```
Comment (keszybz) [inline src/test/test-label.c:26]: Please add a comment explaining this "40".

File: src/test/test-label.c [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+#include <errno.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "fs-util.h"
+#include "label.h"
+#include "path-util.h"
+#include "string-util.h"
+#include "tests.h"
+
```
Comment (keszybz) [inline src/test/test-label.c:16]: Unnecessary newline here.

File: src/test/test-label.c [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        label_ops_reset();
+
+        assert_se(label_ops_set(&test_label_ops) == 0);
+        /* attempt to reset label_ops when already set */
+        assert_se(label_ops_set(&test_label_ops) == -EBUSY);
+}
+
+TEST(label_ops_pre) {
+
```
Comment (keszybz) [inline src/test/test-label.c]: Unnecessary newline here.

File: src/test/test-label.c [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        label_ops_set(&test_label_ops);
+
+        assert_se(label_ops_pre(1, "/abcd", 0755) == 0);
+        assert_se(label_ops_pre(1, "/restricted_directory", 0755) == -EACCES);
+        assert_se(label_ops_pre(2, "abcd", 0644) == -EINVAL);
+        assert_se(label_ops_pre(1, "/wekrgoierhgoierhqgherhgwklegnlweehgorwfkryrit", 0755) == -ENAMETOOLONG);
+}
+
+TEST(label_ops_post) {
+
```
Comment (keszybz) [inline src/test/test-label.c]: Unnecessary newline here.

File: src/test/test-label.c [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        static const LabelOps test_label_ops = {
+                .pre = NULL,
+                .post = post_labelling_func,
+        };
+        label_ops_reset();
```
Comment (keszybz) [inline src/test/test-label.c:135]: Newline is missing here.

File: src/test/test-label.c [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+static int post_labelling_func(int dir_fd, const char *path) {
+       int r;
+
+        /* assume label policies that restrict certain labels */
+        r = check_path(dir_fd, path);
+        if (r < 0)
+                return r;
+
+        /* custom post labelling logic */
+        return 0; /*on sucess*/
```
Comment (keszybz) [inline src/test/test-label.c]: Just `return check_path(dir_fd, path)` ?

File: src/test/test-label.c [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+        assert(mode != MODE_INVALID);
+
+        /* sample pre_labelling task */
+        r = check_path(dir_fd, path);
+        if (r < 0)
+                return r;
+
+        /* custom pre labelling logic here */
+        return 0; /* on success */
```
Comment (keszybz) [inline src/test/test-label.c]: Just `return check_path(dir_fd, path)` ?

**Review by keszybz — CHANGES_REQUESTED** [review]

**Comment by devdekunle** [comment]
@keszybz Good morning. Please I made changes as you requested while I added more functionality. The pre_labelling_func and post_labelling_func are custom functions. I also have an issue. The unit tests pass from my local tests but some checks on the ci are failing such the mkosi/ci and unittest with Asan and ubsan specified. Please can you guide me on what these checks are testing for because I can't see any failure from my local machine. Maybe there's something I'm not checking.

**Comment by devdekunle** [comment]
@bluca @keszybz. I have successfully fixed the errors after lots of debugging and persistence. Please kindly review. Thank you.

**Review by bluca — APPROVED** [review]

**Pull request merged**